### ISSUE TITLE
hyperas score

### ISSUE THREAD

**Issue by jesuisnicolasdavid** [maxpumperla/hyperas#4]
I am trying to see different score after the final step of hyperas, is there a way to return a y_pred from the function to use in the main() ? 

> from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score, confusion_matrix, roc_curve, roc_auc_score
> from matplotlib import pyplot as plt
> %matplotlib inline
> from __future__ import print_function
> from hyperopt import Trials, STATUS_OK, tpe
> from hyperas import optim
> from hyperas.distributions import choice, uniform
> 
> 
> def keras_model():
>     from keras.preprocessing import sequence
>     from keras.models import Sequential
>     from keras.layers.core import Dense, Dropout, Activation
>     from keras.layers.embeddings import Embedding
>     from keras.layers.recurrent import LSTM
>     from keras.datasets import imdb
>     from keras.callbacks import EarlyStopping, ModelCheckpoint
> 
>     max_features = 20000
>     maxlen = 100
> 
>     print('Loading data...')
>     (X_train, y_train), (X_test, y_test) = imdb.load_data(nb_words=max_features, test_split=0.2)
>     print(len(X_train), 'train sequences')
>     print(len(X_test), 'test sequences')
> 
>     print("Pad sequences (samples x time)")
>     X_train = sequence.pad_sequences(X_train, maxlen=maxlen)
>     X_test = sequence.pad_sequences(X_test, maxlen=maxlen)
>     print('X_train shape:', X_train.shape)
>     print('X_test shape:', X_test.shape)
> 
>     print('Build model...')
>     model = Sequential()
>     model.add(Embedding(max_features, 128, input_length=maxlen))
>     model.add(LSTM(128))
>     model.add(Dropout({{uniform(0, 1)}}))
>     model.add(Dense(1))
>     model.add(Activation('sigmoid'))
> 
>     model.compile(loss='binary_crossentropy',
>                   optimizer='adam',
>                   class_mode="binary")
> 
>     early_stopping = EarlyStopping(monitor='val_loss', patience=4)
>     checkpointer = ModelCheckpoint(filepath='keras_weights.hdf5',
>                                    verbose=1,
>                                    save_best_only=True)
> 
>     hist = model.fit(X_train, y_train,
>                      batch_size={{choice([32, 64, 128])}},
>                      # batch_size=128,
>                      nb_epoch=1,
>                      validation_split=0.08,
>                      show_accuracy=True,
>                      callbacks=[early_stopping, checkpointer])
> 
>     score, acc = model.evaluate(X_test, y_test, show_accuracy=True, verbose=0)
> 
>     y_pred = model.predict_classes(X_test)
> 
>     print ('Accuracy:', accuracy_score(y_test, y_pred))
>     print ("Precision", precision_score(y_test, y_pred))
>     print ("Recall", recall_score(y_test, y_pred))
>     print ("f1_score", f1_score(y_test, y_pred))
>     print ("confusion_matrix")
>     print (confusion_matrix(y_test, y_pred))
>     fpr, tpr, tresholds = roc_curve(y_test, y_pred)
>     auc = roc_auc_score(y_test, y_pred)
> 
>     #Courbe
>     plt.figure(figsize=(6, 4))
>     plt.plot(fpr, tpr, label='ROC Curve. AUC = {0:.2f}'.format(auc))
>     plt.plot([0, 1], [0, 1], 'r--')
>     plt.xlim([0.0, 1.0])
>     plt.ylim([0.0, 1.0])
>     plt.xlabel('FPR')
>     plt.ylabel('TPR')
>     plt.title('ROC Curve')
>     plt.legend(loc='lower right')
>     #print('Test accuracy:', acc)
> 
>     return {'loss': -acc, 'status': STATUS_OK}
> 
> if __name__ == '__main__':
>     best_run = optim.minimize(keras_model,
>                               algo=tpe.suggest,
>                               max_evals=1,
>                               trials=Trials())
>     print(best_run)

**Comment by maxpumperla** [comment]
Generally, the idea of hyperopt is to find the best performing model and return its hyperparameters. After that you get back to whatever you were trying to solve in the first place. In your case that means you plug in those parameters to your model, train it on the full data set and evaluate as you wish. That is what you can do right now.

I certainly do see the need for an example for this, though, i.e. how do I return the already trained, best model. Depending on time constraints, I hope to add something like this in the coming two weeks.

**Comment by jesuisnicolasdavid** [comment]
Well ok that's very nice ! Hope to see this new feature soon ! Thanks for your answer !

**Comment by maxpumperla** [comment]
Thanks, let's leave this open to pressure myself to actually do it. :)

**Comment by jesuisnicolasdavid** [comment]
Yes ! That's awesome ! Thank you!

**Comment by maxpumperla** [comment]
Alright! Note that you have to work on master for now to use this. I make a new release tonight.
